pkg/operator/controller/dns/controller.go

```diff
@@ -275,7 +285,13 @@ func (r *reconciler) publishRecordToZones(zones []configv1.DNSZone, record *iov1
 			LastTransitionTime: metav1.Now(),
 		}
 
-		if recordIsAlreadyPublishedToZone(record, &zone) {
+		if dnsPolicy != iov1.ManagedDNS {
```

Comment:
@gcs278 I'm planning to refactor this function to make it less verbose and also thinking of modifying the behaviour where the status would have all the condition types always. Meaning, upon
- successfully publishing the record: `Published=True` and `Failed=False`
- when failed to publish the record: `Publish=False` and `Failed=True`
- when unmanaged: `Published=False` and `Failed=False`

This would simplify the `computeDNS()` where we wouldn't have to sort and probably can reuse the previous function. WDYT?

Reply:
That sounds much better to me if you can remove the logic of sorting and checking the last condition. I'm using this pattern of always including all conditions with either true or false in my PR (https://github.com/openshift/cluster-ingress-operator/pull/769). So I think it's a valid idea.

Reply:
Interesting pattern, might use it in the future. So post [this](https://coreos.slack.com/archives/CE4L0F143/p1659622155245749?thread_ts=1659356092.005309&cid=CE4L0F143), we will not need to complicate this further. It'll mostly go back to the previous implementation.